cpp/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2017 Snowflake Computing, Inc. All rights reserved.
+#
+# CMakeList for Snowflake Client++
+#
+
+set(SOURCE_FILES_CXX
+        include/snowflake++/SnowflakeInclude.hpp
+        include/snowflake++/SnowflakeConnection.hpp
+        include/snowflake++/SnowflakeStatement.hpp
+        lib/SnowflakeConnection.cpp
+        lib/SnowflakeStatement.cpp)
+
+set(TESTLIB_OPTS_CXX snowflakeclient -Wl,--whole-archive curl ssl crypto pthread aws-cpp-sdk-core aws-cpp-sdk-s3 -Wl,--no-whole-archive)
+
+add_library(snowflakeclient++ STATIC ${SOURCE_FILES_CXX})
```

Comment:
@smtakeda should the library name also change to snowflakeclientcpp?

Reply:
Ideally one library includes both c and C++ API. I believe a C application could link properly as the linker picks right objects from *.a file.